### PR TITLE
Update Postgresql chart's version from ~9 to ~11

### PR DESCRIPTION
**Pull request by amitza**
## Summary
Updating the dependencies of the main prefect server chart in order to fix https://github.com/PrefectHQ/server/issues/374

## Importance
The Postgres' current version (9.3.2) isn't accessible through the [Bitnami chart repo](https://charts.bitnami.com/bitnami), the only version there is 11+.

## Checklist

This PR:

- [ ] adds new tests (if appropriate)
- [x] adds a change file in the `changes/` directory (if appropriate)

### CODE REVIEW

**Comment by madkinsz** [comment]
Hi! I see a similar fix at https://github.com/PrefectHQ/server/pull/376 but yours includes some additional changes. Could you explain those?

**Comment by jawnsy** [comment]
Thanks so much for your contribution!

Please make sure that you test the case of upgrading (installing an old version of the chart, upgrading to this one).

I think changing PostgreSQL versions may cause problems when upgrading, because I think there's a manual upgrade step required at the PostgreSQL level - typically taking a backup and restoring: https://www.postgresql.org/docs/14/upgrading.html

The chart also has some documentation describing an upgrade process: https://docs.bitnami.com/kubernetes/infrastructure/postgresql/administration/upgrade/

**Comment by amitza** [comment]
I managed to update the postgresql chart and helm-install it on my KinD cluster.
I'm not sure why the tests fail.

File: .circleci/config.yml [hunk 1]
```diff
@@ -58,21 +58,22 @@ executors:
           PREFECT_SERVER__DATABASE__USERNAME: prefect
           PREFECT_SERVER__DATABASE__PASSWORD: pr343ct
 
-      - image: "postgres:11"
+      - image: "bitnami/postgresql:14.4.0-debian-11-r9"
```
Comment (madkinsz) [inline .circleci/config.yml:61]: Hi! Curious why you made changes to the CircleCI config

**Comment by madkinsz** [comment]
@amitza I do not have permissions to push to your branch so I cannot update this with the CI fixes in `master`. 

I'm willing to accept this if you revert your changes to the CircleCI config.

File: .circleci/config.yml (hunk 1 above)
Comment (amitza) [reply .circleci/config.yml]: I've reverted the postgres version

Comment (madkinsz) [reply .circleci/config.yml]: Thanks! You may need to revert the other changes as well.

If you merge update with `master` this should be able to pass CI.

**Comment by madkinsz** [comment]
Thanks for contributing! This is complete in #384

**Pull request merged**